src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,99 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static RegistrationService service;
+    private User user;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+    }
+
+    @Test
+    void addNull_NotOk() {
+        user = null;
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "User can't be null");
+    }
+
+    @Test
+    void addNegativeAge_NotOk() {
+        user.setLogin("Jack");
+        user.setPassword("qwertyu");
+        user.setAge(-1);
+        assertThrows(RuntimeException.class, () ->
+                 service.register(user), "Age can't be negative");
+    }
+
+    @Test
+    void addNullLogin_NotOk() {
+        user.setLogin(null);
+        user.setPassword("qwertyu");
+        user.setAge(21);
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "Login can't be null");
+    }
+
+    @Test
+    void addNullPassword_NotOk() {
+        user.setLogin("Jack");
+        user.setPassword(null);
+        user.setAge(21);
+        assertThrows(RuntimeException.class, () ->
+                service.register(user), "Password can't be null");
+    }
+
+    @Test
+    void userIsTooYoung_NotOk() {
+        user.setLogin("Jack");
+        user.setPassword("qwertyu");
+        user.setAge(10);
```

Comment:
can we put initialization in setUp method like you do in storage test and just change necessary fields for test purposes?